webapp/lib/model.g.dart

```diff
@@ -371,7 +372,8 @@ class Tag {
       ..text = String_fromData(data['text'])
       ..type = TagType.fromString(data['type'] as String) ?? TagType.Normal
       ..shortcut = String_fromData(data['shortcut'])
-      ..filterable = bool_fromData(data['filterable']);
+      ..filterable = bool_fromData(data['filterable'])
+      ..category = String_fromData(data['category']);
```

Comment:
Does it make sense for `category` to always have a non-null value?
If so... you could append `?? ''` to this line and remove null checks elsewhere.

Reply:
Yes, I think it does. And might make sense to have done the same for `filterable`. I'll follow up with another PR for this. thanks @danrubel !

Speaking of - at the moment I don't think the UI has any checks as to what is optional in the data structure (and so can have a default value) and what is required. I believe that we're primarily relying on the backend infrastructure to ensure the data is in the right format. Would it make sense to add some of those checks on the UI side as well? @danrubel @lukechurch what do you think?

Reply:
@marianamarasoiu Good question. We can easily add default values to replace nulls or missing data fields and there is some backend to UI data coercion in place (e.g. [int_fromData](https://github.com/larksystems/nook/blob/d1c902201d85023bd6c5c6f6c3911e850a7fa84c/webapp/lib/model.g.dart#L536)), but you are right that we don't have any UI checks on data sent to the backend other than Dart types themselves.

@lukechurch Thoughts?

Reply:
I think it makes sense for us to add verification checks on the importing of the data - mainly to improve the error reporting and localisation. I'm not sure that we want the UI trying to do much repair, but checking makes sense.